reconcile/aus/base.py

```diff
@@ -95,36 +102,146 @@ def get_ocm_env_upgrade_specs(
         ...
 
 
-# consider first lower versions and lower soakdays (when versions are equal)
-def sort_key(d: dict) -> tuple:
-    return (
-        parse_semver(d["current_version"]),
-        d["conditions"].get("soakDays") or 0,
-    )
+class GateAgreement(BaseModel):
+    id: str
+
+    def create(self, ocm: OCM, cluster_name: str) -> None:
+        agreement = ocm.create_version_agreement(self.id, cluster_name)
+        if agreement.get("version_gate") is None:
+            logging.error(
+                f"Unexpected response while creating version "
+                f"agreement with id {self.id} for cluster {cluster_name}"
+            )
+
+
+class AbstractUpgradePolicy(ABC, BaseModel):
```

Comment:
i'm a bit confused why this contains an `action`. so it is more than a policy but also instructions on what to do with it (create, delete)?

Reply:
I agree, will move this portion into a separate class